python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -163,28 +163,53 @@ def location(self):
             tmpLocations = self.locations
         return tmpLocations
 
-    def get(self):
+    def get(self, localPath=''):
         """
         Retrieves locally all files matching this MassStorageFile object pattern
         """
+        self._actual_copy()
 
-        to_location = self.localDir
-
-        if not os.path.isdir(self.localDir):
-            if self._getParent() is not None:
-                to_location = self.getJobObject().outputdir
-            else:
-                logger.error("%s is not a valid directory.... Please set the localDir attribute" % self.localDir)
-                return
+    def _actual_copy(self, localPath=''):
+        """
+        Copy a the file to the local storage using the get mechanism
+        Args:
+            localPath (str): Target path where the file is to copied to
+        """
+        if not localPath:
+            to_location = self.localDir
+            # FIXME? THIS IS POTENTIALLY VERY, VERY UNEXPECTED/UNDOCUMENTED BEHAVIOUR
+            if not os.path.isdir(to_location):
+                if self._getParent() is not None:
+                    try:
+                        new_dir = self.getJobObject().outputdir
+                        if to_location:
+                            logger.warning("Setting get path of file to be: '%s' from: '%s'" % (new_dir, to_location))
+                        to_location = new_dir
+                    except AssertionError:
```

Comment:
`getJobObject()` this shouldn't happen but the getter in the Job is/was a pain with regard to outputfiles.